### PR TITLE
add sample links

### PR DESCRIPTION
**Pull request by yiliuTo**
1. Add sample project urls from repo https://github.com/Azure-Samples/azure-spring-boot-samples of each spring libraries.
2. Update the latest release version.

### CODE REVIEW

**Comment by yiliuTo** [comment]
Hi @wangmingliang-ms and @backwind1233 , please help to review this pr, thanks.

File: sdk/spring/spring-reference.yml [hunk 1]
```diff
@@ -481,14 +496,15 @@
       artifacts:
         - artifactId: azure-spring-cloud-starter-eventhubs-kafka
           groupId: com.azure.spring
-          versionGA: 2.4.0
+          versionGA: 2.7.0
           versionPreview: 2.0.0-beta.1
           description: Microsoft's Spring Cloud Stream Binder helps developers to finish the auto-configuration of Event Hubs with Kafka protocol.
           type: spring
           links:
             repopath: https://search.maven.org/artifact/com.azure.spring/azure-spring-cloud-starter-eventhubs
```
Comment (backwind1233) [inline sdk/spring/spring-reference.yml:504]: is the link right?

Comment (yiliuTo) [reply sdk/spring/spring-reference.yml]: done

**Comment by wangmingliang-ms** [comment]
LGTM

**Review by backwind1233 — APPROVED** [review]
LGTM

**Pull request merged**